api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ViralLoadIntensiveMonitoringCohortQueries.java

```diff
@@ -0,0 +1,2685 @@
+    map.put("53", hivMetadata.getMasterCardEncounterType().getEncounterTypeId());
+    map.put("856", hivMetadata.getHivViralLoadConcept().getConceptId());
+
+    String sql =
+        " SELECT p.patient_id FROM   patient p "
+            + "       INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "       INNER JOIN ((SELECT result.patient_id, MIN(result.vl_date) AS vl_dateP  FROM( "
+            + "           SELECT p.patient_id, e.encounter_datetime AS vl_date "
+            + "                   FROM   patient p "
+            + "                          INNER JOIN encounter e "
+            + "                                  ON e.patient_id = p.patient_id "
+            + "                          INNER JOIN obs o "
+            + "                                  ON o.encounter_id = e.encounter_id "
+            + "                   WHERE  p.voided = 0 "
+            + "                          AND e.voided = 0 "
+            + "                          AND o.voided = 0 "
+            + "                          AND e.encounter_type = ${6} "
+            + "                          AND o.concept_id = ${856} "
+            + "                          AND o.value_numeric >= 1000 "
+            + "                          AND e.location_id = :location "
+            + "                          AND e.encounter_datetime >= :evaluationPeriodStartDate "
+            + "                          AND e.encounter_datetime <= :evaluationPeriodEndDate "
+            + " "
+            + "                   UNION "
+            + " "
+            + "                   SELECT p.patient_id, o.obs_datetime AS vl_date "
+            + "                   FROM   patient p "
+            + "                          INNER JOIN encounter e "
+            + "                                  ON e.patient_id = p.patient_id "
+            + "                          INNER JOIN obs o "
+            + "                                  ON o.encounter_id = e.encounter_id "
+            + "                   WHERE  p.voided = 0 "
+            + "                          AND e.voided = 0 "
+            + "                          AND o.voided = 0 "
+            + "                          AND e.encounter_type = ${53} "
+            + "                          AND o.concept_id = ${856} "
+            + "                          AND o.value_numeric >= 1000 "
+            + "                          AND e.location_id = :location "
+            + "                          AND o.obs_datetime >= :evaluationPeriodStartDate "
+            + "                          AND o.obs_datetime <= :evaluationPeriodEndDate) AS result  "
+            + "                          GROUP BY result.patient_id))AS result_vl "
+            + "               ON result_vl.patient_id = p.patient_id "
+            + "        WHERE  e.encounter_type = ${35} "
```

Comment:
its correct because the spec says:
All patients with 3 APSS/PP consultations (encounter type 35) within 99 days of first VL date >= 1000 (check Y section: if VL is from encounter type 6 use encounter datetime, if VL is from encounter type 53 use obs datetime) as VL dateP as follows:
One APSS/PP (encounter type 35) in the same day as VL dateP